packages/block-editor/src/components/inserter/hooks/use-insertion-point.js

```diff
@@ -86,10 +87,23 @@ function useInsertionPoint( {
 		insertBlocks,
 		showInsertionPoint,
 		hideInsertionPoint,
-	} = useDispatch( blockEditorStore );
+		setLastFocus,
+	} = unlock( useDispatch( blockEditorStore ) );
 
 	const onInsertBlocks = useCallback(
 		( blocks, meta, shouldForceFocusBlock = false ) => {
+			// When we are trying to move focus or select a new block on insert, we also
+			// need to clear the last focus to avoid the focus being set to the wrong block
+			// when tabbing back into the canvas if the block was added from outside the
+			// editor canvas.
+			if (
+				shouldForceFocusBlock ||
+				shouldFocusBlock ||
+				selectBlockOnInsert
+			) {
+				setLastFocus( null );
```

Comment:
I considered doing this within the inserter sidebar menu, but I think it should live here as it will be an issue for anyone trying to insert new blocks from outside the editor canvas. This allows people to choose if they want focus to return to the canvas where it was (preserve last focus) or clear it and focus the new block.